react-app/src/components/ConfigForm.js

```diff
@@ -1,7 +1,58 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Alert, Button } from 'react-bootstrap';
 
-function ConfigForm() {
-  return <p className="page-text text-centered">The config editor form will display here.</p>;
+function ConfigForm(props) {
+  const [showSavedAlert, setShowSavedAlert] = useState(false);
+  const [savedMessage, setSavedMessage] = useState('');
+  const [showErrorAlert, setShowErrorAlert] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  function onSaveAs() {
+    window.api
+      .saveConfigAs(props.schema)
+      .then((result) => {
+        if (result !== null) {
+          // if saveOutput() returns true, then the save process succeeded
+          setShowSavedAlert(true);
+          setSavedMessage('Files saved to '.concat(result));
+        }
+        // If result is null, the process was cancelled, and nothing should be done.
+      })
+      .catch((error) => {
+        setShowErrorAlert(true);
+        setErrorMessage(error.message);
+        setShowSavedAlert(false);
+      });
+  }
+  function onBack() {
+    props.setShowForm(false);
+  }
+  return (
+    <div className="page-container">
```

Comment:
I don't know if we want to use the page-container here. The CSS says this container will always be 100vh; since the config form renders below the title of the configEditor page, the result is that the buttons fall off screen. I've included a snapshop that attempts to show this. 
![image](https://user-images.githubusercontent.com/5703736/126377787-d36205a0-5755-4afd-9cdb-b964854d7fa6.png)

Reply:
I wound up having to use a state variable to control which class was used for the div surrounding the ConfigEditor component. When the form is displayed, the page as a whole needs to have a height 100vh for the spacing to work properly; when there's no form, the 100vh messes up the button placement.